mypyc/test-data/irbuild-classes.test

```diff
@@ -460,18 +460,20 @@ def f(x: A) -> B:
 def f(x):
     x :: __main__.A
     r0 :: object
-    r1 :: bool
-    r2, r3 :: __main__.B
+    r1 :: ptr
+    r2 :: bool
+    r3, r4 :: __main__.B
 L0:
     r0 = __main__.B :: type
-    r1 = type_is x, r0
-    if r1 goto L1 else goto L2 :: bool
+    r1 = get_element_ptr x ob_type :: PyObject
+    r2 = r1 == r0
```

Comment:
You should read the value of the struct item before comparing it here.

Reply:
> You should read the value of the struct item before comparing it here.

I don't think so. The `Py_TYPE` macro is just getting the struct item(a pointer) and compare with another pointer.

Reply:
Yeah, but this code gets a pointer to the struct item, not the actual struct item? We need to `LoadMem` to read the value.

Here's how it's defined:
```
#define Py_TYPE(ob)             (((PyObject*)(ob))->ob_type)
```

Note that it reads the value of `ob_type`, it doesn't just take a pointer to it.